### PR TITLE
BUG:  set value with iloc and list like indexer

### PR DESCRIPTION
**Pull request by erfannariman**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Review by jreback — CHANGES_REQUESTED** [review]
what issue is this closing? this would need a test; i am not sure why you think this needs changing.

**Comment by erfannariman** [comment]
> what issue is this closing? this would need a test; i am not sure why you think this needs changing.

It's for #36741 , might not be the place to start, that's why its draft

File: pandas/core/internals/blocks.py [hunk 1]
```diff
@@ -932,7 +932,7 @@ def setitem(self, indexer, value):
         # value must be storable at this moment
         if is_extension_array_dtype(getattr(value, "dtype", None)):
             # We need to be careful not to allow through strings that
-            #  can be parsed to EADtypes
```
Comment (jbrockmendel) [inline pandas/core/internals/blocks.py:935]: this is a pattern i use intentionally to distinguish between

```
# this is a a very long comment that is split into
#  multiple lines
# this is a separate comment that happens to be nearby
```

Comment (erfannariman) [reply pandas/core/internals/blocks.py:935]: Got it, reverted the whole file.

**Comment by erfannariman** [comment]
@jbrockmendel this seems to work, but when the dtype of the new values changes this will return the wrong dtype. For example (see example below), although the values will be correct. So it will solve the problem of the reference issue.
```python
df_orig = DataFrame(
    [["1", "2", "3", ".4", 5, 6.0, "foo"]], columns=list("ABCDEFG")
)

df = df_orig.copy()
df.iloc[:, 0:2] = df.iloc[:, 0:2].astype(np.int64)
expected = DataFrame(
    [[1, 2, "3", ".4", 5, 6.0, "foo"]], columns=list("ABCDEFG")
)
tm.assert_frame_equal(df, expected)
```

Any ideas where I might look?

**Comment by simonjayhawkins** [comment]
supersed by #37432

**Pull request closed without merging**